### DISCUSSION TITLE
[11 gennaio 2022 - B] Esercizio 6

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#204] [Esami]
**Esercizio 6 (2+2 punti)**

Dati i numeri in rappresentazione esadecimale IEEE 754 half-precision $X = C500$ e $Y = 4300$,

(a) eseguire l’operazione $X+Y$ secondo lo standard IEEE 754
(b) rappresentare il risultato in notazione esadecimale, poi convertire operandi e risultato in base dieci.

**Comment by Elia-Belli** [comment]
- $X = C500_{16} = 1\ 10001\ 0100000000 = -2^2\ 1.01 $
  $Y = 4300_{16} = 0\ 10000\ 1100000000 = +2^1\ 1.11 = +2^2\ 0.111$
  $- X = 0.10 + 1 = 10.11$ (siccome negativo per la somma converto in Ca2)
 $10.110 +$
 $00.111 =$
 $11.101$
 $-Z = 2^2\ 11.101 \rightarrow -2^2\ 00.011 = -2^0\ 1.1$ (estraggo il segno dalla mantissa passando al complemento e normalizzo)
$Z = 1\ 01111\ 1000000000 = BE00_{16}$

- $X = -5_{10}$
$Y = +3.5_{10}$
$Z = -1.5_{10}$

**Comment by RaulComaz** [comment]
ma se facessi x - y = -z lo consi.derate giusto? tanto il risultato non cambia

**Comment by fraxb04** [comment]
@Elia-Belli 
corretto
